pkg/runtime/runtime.go

```diff
@@ -663,37 +668,24 @@ func (a *DaprRuntime) subscribeTopic(parentCtx context.Context, name string, top
 
 	ctx, cancel := context.WithCancel(parentCtx)
 	policy := a.resiliency.ComponentInboundPolicy(ctx, name, resiliency.Pubsub)
-	err := a.pubSubs[name].component.Subscribe(ctx, pubsub.SubscribeRequest{
-		Topic:    topic,
-		Metadata: route.metadata,
-	}, func(ctx context.Context, msg *pubsub.NewMessage) error {
-		if msg.Metadata == nil {
-			msg.Metadata = make(map[string]string, 1)
-		}
-
-		msg.Metadata[pubsubName] = name
-
-		rawPayload, err := contribMetadata.IsRawPayload(route.metadata)
-		if err != nil {
-			log.Errorf("error deserializing pubsub metadata: %s", err)
-			if route.deadLetterTopic != "" {
-				if dlqErr := a.sendToDeadLetter(name, msg, route.deadLetterTopic); dlqErr == nil {
-					// dlq has been configured and message is successfully sent to dlq.
-					diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, pubsubName, strings.ToLower(string(pubsub.Drop)), msg.Topic, 0)
-					return nil
-				}
+	var err error
+	routeMetadata := route.metadata
+	if routeMetadata[BulkSubscribe] == "true" {
+		err = a.bulkSubscribeTopic(ctx, policy, name, topic, route)
+	} else {
+		err = a.pubSubs[name].component.Subscribe(ctx, pubsub.SubscribeRequest{
+			Topic:    topic,
+			Metadata: routeMetadata,
+		}, func(ctx context.Context, msg *pubsub.NewMessage) error {
+			if msg.Metadata == nil {
+				msg.Metadata = make(map[string]string, 1)
```

Comment:
Reviewers : the whole normal subscribe logic is moved in else clause AFAIK ..
@DeepanshuA correct me if i am wrong ...

Reply:
Yups, it was. I liked your suggestion offline to have an if block only for `bulkSubscribe` and process and return from there itself. And, instead of else, keep it afterwards - so that any reviewer can easily see that subscribe code is not disturbed due to bulkSubscribe.